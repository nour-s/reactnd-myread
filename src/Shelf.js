import React from "react";
import Book from "./Book";
import PropTypes from "prop-types";

class Shelf extends React.Component {
	render() {
		return (
			<div className="bookshelf">
				<h2 className="bookshelf-title">{this.props.shelf.name}</h2>
				<div className="bookshelf-books">
					<ol className="books-grid">
						{this.props.shelf.books.map(book => (
							<li key={book.id}>
								<Book
									book={book}
									onMoveBook={this.props.onMoveBook}
								/>
							</li>
						))}
					</ol>
				</div>
			</div>
		);
	}
}

Shelf.propTypes = {
	shelf: PropTypes.shape({
		books: PropTypes.array
	}),
	onMoveBook: PropTypes.func
};

export default Shelf;
