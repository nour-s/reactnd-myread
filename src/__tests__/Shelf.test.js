import React from "react";
import renderer from "react-test-renderer";
import Shelf from "../Shelf";
import { mount } from "enzyme";

describe("Shelf test", () => {
	let shelf = { id: "shelf 1", books: [{ id: 1, title: "Title 1" }] };
	let onMoveBook;
	let rawComponent;

	beforeEach(() => {
		onMoveBook = jest.fn();
		rawComponent = <Shelf shelf={shelf} onMoveBook={onMoveBook} />;
	});

	it("Shelf should match snapshot", () => {
		const component = renderer.create(rawComponent);
		const tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});

	it("Shelf should render the proper number of Book component", () => {
		const wrapper = React.createElement(() => rawComponent);
		let component = mount(wrapper);
		expect(component.find("Book")).toHaveLength(1);
	});

	it("Shelf should pass onMoveBook function to books", () => {
		const wrapper = React.createElement(() => rawComponent);
		let component = mount(wrapper);

		component.find("Book").forEach(b => {
			expect(b.prop("onMoveBook")).toBe(onMoveBook);
		});
	});
});
