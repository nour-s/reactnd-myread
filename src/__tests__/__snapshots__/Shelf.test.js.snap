// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shelf test Shelf should match snapshot 1`] = `
<div
  className="bookshelf"
>
  <h2
    className="bookshelf-title"
  />
  <div
    className="bookshelf-books"
  >
    <ol
      className="books-grid"
    >
      <li>
        <div
          className="book"
        >
          <div
            className="book-top"
          >
            <div
              className="book-cover"
              style={
                Object {
                  "backgroundImage": "url()",
                  "height": 193,
                  "width": 128,
                }
              }
            />
            <div
              className="book-shelf-changer"
            >
              <select
                onChange={[Function]}
                value="none"
              >
                <option
                  disabled={true}
                  value="move"
                >
                  Move to...
                </option>
                <option
                  disabled={false}
                  value="currentlyReading"
                >
                  Currently Reading
                </option>
                <option
                  disabled={false}
                  value="read"
                >
                  Read
                </option>
                <option
                  disabled={false}
                  value="none"
                >
                  None
                </option>
              </select>
            </div>
          </div>
          <div
            className="book-title"
          >
            Title 1
          </div>
        </div>
      </li>
    </ol>
  </div>
</div>
`;
